ru/turing/courses/lesson3/VersionA/Registry.java

```diff
@@ -0,0 +1,252 @@
+package ru.turing.courses.lesson3.VersionA;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+
+public class Registry {
+    public static class Animal {
+        private Map<String, Animal> storage = new HashMap<>();
+
+        /**
+         * add(name, animal)
+         * добавить в реестр новую запись
+         *
+         * @param name
+         * @param animal
+         */
+        public void add(String name, Animal animal) {
+            storage.put(name, animal);
+        }
+
+        /**
+         * removeByKey(String name)
+         * удалить из реестра запись по заданному ключу. Возвращать должен значение записи
+         *
+         * @param name
+         * @param <T>
+         * @return
+         */
+        public <T> T removeByKey(String name) {
```

Comment:
```suggestion
        public T removeByKey(String name) {
```
Здесь и далее явное объявление дженериков не нужно: оно прописано при объявлении регистра. Здесь же просто указываем тип возвращаемого значения.